test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_work_groups.cpp

```diff
@@ -0,0 +1,274 @@
+                                std::vector<cl_int> &resultData,
+                                size_t old_global_size = 0)
+    {
+        cl_int error;
+        cl_mutable_dispatch_config_khr dispatch_config{
+            CL_STRUCTURE_TYPE_MUTABLE_DISPATCH_CONFIG_KHR,
+            nullptr,
+            command,
+            0, // num_args
+            0, // num_svm_arg
+            0, // num_exec_infos
+            0, // work_dim (0 means no change to dimensions)
+            nullptr, // arg_list
+            nullptr, // arg_svm_list (nullptr means no change)
+            nullptr, // exec_info_list
+            nullptr, // global_work_offset
+            &new_global_size, // global_work_size
+            nullptr // local_work_size
+        };
+
+        cl_mutable_base_config_khr mutable_config{
+            CL_STRUCTURE_TYPE_MUTABLE_BASE_CONFIG_KHR, nullptr, 1,
+            &dispatch_config
+        };
+
+        error = clUpdateMutableCommandsKHR(command_buffer, &mutable_config);
+        test_error(error, "clUpdateMutableCommandsKHR failed");
+
+        clEventWrapper events[2];
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &events[0]);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &events[0]);
+        test_error(error, "clWaitForEvents failed");
+
+        error = clEnqueueReadBuffer(queue, stream, CL_FALSE, 0, sizeToAllocate,
+                                    resultData.data(), 0, nullptr, &events[1]);
+        test_error(error, "clEnqueueReadBuffer failed");
+
+        error = clWaitForEvents(1, &events[1]);
+        test_error(error, "clWaitForEvents failed");
+
+        size_t expected_groups = new_global_size / local_work_size;
+        size_t old_num_of_groups = old_global_size / local_work_size;
+        for (size_t i = 0; i < global_work_size; ++i)
+        {
+            if (i >= new_global_size && old_num_of_groups != resultData[i])
+            {
+                log_error("Data failed to verify: old_num_of_groups != "
+                          "resultData[%zu]=%d\n",
+                          i, resultData[i]);
+                return TEST_FAIL;
+            }
+            else if (i < new_global_size && expected_groups != resultData[i])
+            {
+                log_error("Data failed to verify: expected_groups != "
+                          "resultData[%zu]=%d\n",
+                          i, resultData[i]);
+                return TEST_FAIL;
+            }
+        }
+
+        return CL_SUCCESS;
+    }
+
+    size_t info_global_size = 0;
+    static constexpr size_t test_global_work_size = 64;
+    static constexpr size_t update_global_size = 16;
+    const size_t local_work_size = 8;
+    const size_t sizeToAllocate = 64 * sizeof(cl_int);
+    cl_mutable_command_khr command = nullptr;
+    clCommandBufferWrapper command_buffer;
```

Comment:
This variable is shadowing the base class `command_buffer` member, we therefore use it in the tests without initializing it

Reply:
I fixed this, thank you